test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetTag.java

```diff
@@ -63,7 +58,22 @@
 //   3. Add tests for hybrid snippets
 
 /*
- * General notes.
+ * General notes
+ * =============
+ *
+ * To simplify maintenance of this test suite, a test name uses a convention.
+ * By convention, a test name is a concatenation of the following parts:
+ *
+ *    1. "test"
+ *    2. ("Positive", "Negative")
+ *    3. ("Inline", "External", "Hybrid")
+ *    4. ("Tag", "Markup")
+ *    5. <custom string>
```

Comment:
I assume the purpose of this naming scheme is to make visible what combinations of features are covered by each test. I'm not sure I would consider this enough benefit to justify method names which are very verbose and hard to read (especially when you don't have fresh memory of the scheme above).

Reply:
When one is planning to have many methods, one should better have a plan for how to maintain them. Such a plan should cover navigation, search, and if growth is foreseen, naming new methods. A naming scheme could in itself be such a plan.

I've tried alternative naming schemes but found this one to work best. It's obviously possible that I missed something. If you have other ideas, please share and let's discuss them.

Reply:
> I assume the purpose of this naming scheme is to make visible what combinations of features are covered by each test. I'm not sure I would consider this enough benefit to justify method names which are very verbose and hard to read (especially when you don't have fresh memory of the scheme above).

You are right when saying that this naming scheme helps to survey the test landscape. This is especially helpful while mass-developing tests, which is what I'm doing at the moment.

Reply:
Do you think we could add an underscore here and there to improve readability? For example, we could separate the first four parts of the name from the last one, which is a custom string:
```
testNegativeExternalTag_FileNotFoundModuleSourcePath
```
I'm not against "camelCase", but I have to admit that "snake_case" or a reasonable mix of both styles might work better for longer names.

Reply:
In general, given the size of the file, I like the verbose names, if only because it gives the reader a strong hint up front about the form (nature) of the test.

That being said, now that you have introduced `SnippetTester` (which is great!) you could split this file. I would definitely group positive and negative test cases in the same file, but you could maybe separate `Tag` and `Markup` tests into separate files. (Just a suggestion).   

FWIW, my experience is that it is easier to test/debug files with fewer methods in, although it is on my to-do list to make it easier to run individual test cases.